### PR TITLE
Fixes for podlators for z/OS and for generic EBCDIC

### PR DESCRIPTION
**Pull request by khwilliamson**
It turns out that z/OS can operate in both ASCII and EBCDIC mode.  Previously it was assumed by Perl to be EBCDIC-only.  Some IBMers contributed code later in the Perl 5.35 development cycle to get the ASCII version working.  podlators is one of the few modules that also made this assumption. about z/OS.   Recently, I became aware of this issue, and so I customized the podlators shipped with the Perl core to include the first commit in this P.R, as I didn't  think there was enough time to get a new release of podlators out before the 5.36 code freeze, and it wouldn't be fair to you to ask for such a quick turn-around.

podlators still doesn't fully work on EBCDIC; the core test suite skips its tests.  Part of that is Pod::Simple, which needs fixes first.  But in auditing the code some time ago, I found some areas that made assumptions that the character set is ASCII.  I've been waiting to get all of podlators working before submitting a patch to you with these minor things.  But since the podlators version we ship already needed to be customized, I threw those in too.

The bottom line is that, unless things get reverted for some reason, the podlators shipped with v5.36 Perl will contain these two patches.

The character set translation code in the second patch is completely optimized out on an ASCII machine, so there is no performance penalty for it.

When testing these patches on z/OS, I needed more info as to what is failing, so extra information about the failure is output in encode.t

### CODE REVIEW

**Comment by rra** [comment]
This is great, thanks so much for the PR! podlators is long overdue for another release and I hope to get to that soon.
